processing/src/main/java/io/druid/query/filter/SelectorDimFilter.java

```diff
@@ -225,4 +255,18 @@ private void initLongValue()
       longsInitialized = true;
     }
   }
+
+  private void initFloatValue()
+  {
+    if (floatsInitialized) {
+      return;
+    }
+    synchronized (initLock) {
+      if (floatsInitialized) {
+        return;
+      }
+      valueAsFloat = Floats.tryParse(value);
```

Comment:
If caching resulting predicate directly instead of `valueAsFloat` (it's possible because predicates are immutable), `floatsInitialized` is not needed. Also, locking could be removed and benign race allowed.